### ISSUE TITLE
Feedback - Week 6 Assignment

### ISSUE THREAD

**Issue by ntuvera** [shaankastuar/week6HW#1]
Hey @shaankastuar, I cloned your repo and took a look at your giphy app.

We're missing some functionality here.  You've got the ability to add buttons and display gifs which is a good start.  

Do you think you can add the code necessary to validate empty inputs, so I can't create blank buttons and/or create duplicate buttons?  From a UX point of view, this should be fixed for the page to be considered done.  Also you may want to hit it up with some styling.

Also we're missing some file structure.  Your `hw.js` should be in an `/assets/javascript` folder.  It's only two files for functionality right now, but I want to see proper folder/file structure.  It will be important in the long run, so start now.  Also in terms of file structure, why are we using inline javascript if we already have and `src="hw.js"`?

Your buttons are only displaying one set of gifs.  Look again at your renderButtons function, double check what ln 56-59 are doing.  Then reevaluate your `$('#animalButtons').on('click'`, that is literally saying when I click the div in which the buttons are located.  That div also does not have a data-animal name, neither do you buttons.

You've got some of the functionality we're working on.  It looks like you're able to hit the API with `$.ajax` and manipulate the data, but it still needs fixing as it's pretty buggy.

Also remove debuggers before you submit up, or atleast comment them out.  didn't read your comments until after I wrote all of the above.  Let's talk it out in class, but look at ln 56-59, and 65 to find out why your images are always the same.

**Comment by shaankastuar** [comment]
Yeah, I was going crazy with the ajax calls because no matter what I did,
it kept calling the same GIFs. It wasn't putting it into the URL that was
being sent over, and I kept getting a null and blank.

This one is messed up, I'm going to spend a bunch of tomorrow fixing up
this HW along with some of my previous ones. I haven't updated many of my
repos with the latest files that I have on my computer.

I've also always kept my code in one file (even though I know it's wrong),
as early on I was having trouble getting the file to actually reference
another file with separate styling and code, and I guess I just got in the
habit of keeping it on one page. So I'll make sure not to do that from here
on out and figure out why it wasn't working previously.

On Mon, Mar 21, 2016 at 3:19 AM, Nate Tuvera notifications@github.com
wrote:

> Hey @shaankastuar https://github.com/shaankastuar, I cloned your repo
> and took a look at your giphy app.
> 
> We're missing some functionality here. You've got the ability to add
> buttons and display gifs which is a good start.
> 
> Do you think you can add the code necessary to validate empty inputs, so I
> can't create blank buttons and/or create duplicate buttons? From a UX point
> of view, this should be fixed for the page to be considered done. Also you
> may want to hit it up with some styling.
> 
> Also we're missing some file structure. Your hw.js should be in an
> /assets/javascript folder. It's only two files for functionality right
> now, but I want to see proper folder/file structure. It will be important
> in the long run, so start now. Also in terms of file structure, why are we
> using inline javascript if we already have and src="hw.js"?
> 
> You're buttons are only displaying one set of gifs. Look again at your
> renderButtons function, double check what ln 56-59 are doing. Then
> reevaluate your $('#animalButtons').on('click', that is literally saying
> when I click the div in which the buttons are located. That div also does
> not have a data-animal name, neither do you buttons.
> 
> You've got some of the functionality we're working on. It looks like
> you're able to hit the API with $.ajax and manipulate the data, but it
> still needs fixing as it's pretty buggy.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> https://github.com/shaankastuar/week6HW/issues/1

## 

Shaan Kastuar
shaankastuar@gmail.com
View my LinkedIn Profile
http://www.linkedin.com/profile/view?id=172084583&trk=tab_pro
